File: apps/mybookkeeper/frontend/src/shared/types/document/document.ts
export interface Document {
  id: string;
  user_id: string;
  property_id: string | null;
  created_at: string;
  updated_at: string;
  file_name: string | null;
  file_type: string | null;
  document_type: string | null;
  file_mime_type: string | null;
  email_message_id: string | null;
  external_id: string | null;
  external_source: string | null;
  source: "email" | "upload";
  status: "processing" | "extracting" | "completed" | "failed" | "duplicate" | "deleted";
  error_message: string | null;
  batch_id: string | null;
  is_escrow_paid: boolean;
  deleted_at: string | null;
}

File: apps/mybookkeeper/backend/app/schemas/documents/document.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class DocumentRead(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    property_id: uuid.UUID | None = None
    created_at: datetime
    updated_at: datetime
    file_name: str | None = None
    file_type: str | None = None
    document_type: str | None = None
    file_mime_type: str | None = None
    email_message_id: str | None = None
    external_id: str | None = None
    external_source: str | None = None
    source: str = "upload"
    status: str = "processing"
    error_message: str | None = None
    batch_id: str | None = None
    is_escrow_paid: bool = False
    deleted_at: datetime | None = None

    model_config = {"from_attributes": True}
